pkg/telemetry/vector.go

```diff
@@ -0,0 +1,181 @@
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"text/template"
+
+	"github.com/go-logr/logr"
+	k8ssandraapi "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	telemetry "github.com/k8ssandra/k8ssandra-operator/apis/telemetry/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/cassandra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// Default resources for Vector agent
+	DefaultVectorCpuRequest    = "100m"
+	DefaultVectorMemoryRequest = "128Mi"
+	DefaultVectorCpuLimit      = "2"
+	DefaultVectorMemoryLimit   = "2Gi"
+	DefaultScrapeInterval      = 30
+)
+
+// InjectCassandraVectorAgent adds the Vector agent container to the Cassandra pods.
+// If the Vector agent is already present, it is not added again.
+func InjectCassandraVectorAgent(telemetrySpec *telemetry.TelemetrySpec, dcConfig *cassandra.DatacenterConfig, k8cName string, logger logr.Logger) error {
+	if telemetrySpec != nil {
+		if telemetrySpec.IsVectorEnabled() {
+			logger.Info("Injecting Vector agent into Cassandra pods")
+			vectorImage := defaultVectorImage
+			if telemetrySpec.Vector != nil && telemetrySpec.Vector.Image != "" {
```

Comment:
`telemetrySpec.Vector != nil` is always true here.

Reply:
👍